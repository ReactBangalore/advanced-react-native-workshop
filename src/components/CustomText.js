import React from "react";
import { Text } from "react-native";
export default class HeaderText extends React.Component {
  render() {
    const { header } = this.props;
    const fontFamily = header ? "Roboto" : "serif";
    return <Text style={{ fontFamily }}>{this.props.children}</Text>;
  }
}
