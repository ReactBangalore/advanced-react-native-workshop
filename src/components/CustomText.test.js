import React from "react";
import CustomText from "./CustomText";
import renderer from "react-test-renderer";

it("normal test", () => {
  const rendered = renderer.create(<CustomText />).toJSON();
  expect(rendered).toMatchSnapshot();
});

it("header test", () => {
  const rendered = renderer.create(<CustomText header />).toJSON();
  expect(rendered).toMatchSnapshot();
});
