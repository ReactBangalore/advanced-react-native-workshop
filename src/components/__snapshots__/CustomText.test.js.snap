// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header test 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "fontFamily": "Roboto",
    }
  }
/>
`;

exports[`normal test 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "fontFamily": "serif",
    }
  }
/>
`;
